pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -160,11 +164,6 @@ func (c *VSphereController) sync(ctx context.Context, syncContext factory.SyncCo
 		return err
 	}
 
-	if opSpec.ManagementState != operatorapi.Managed {
-		klog.Warningf("%s: ManagementState is %s, skipping", c.name, opSpec.ManagementState)
-		return nil
-	}
-
```

Comment:
This essentially moves the check from the beginning of `sync()` to the place after infra checks. Is this change relevant to the bug (OCPBUGS-38274) or it's piggybacked? As for me, I'd vote for the previous position of the check: if state!=Managed, we shouldn't care about infra from inside vsphere storage operator.

Reply:
Yeah - this change is intentional. I don't want to be removing stuff if platform is not of `vSphere` type. Though I agree - I do not see how this operator will get "activated" if platform is not vsphere. But - this is more like another failsafe.